trackbase/FairRKPropagator.cxx

```diff
@@ -0,0 +1,625 @@
+  double stepLength = OneStepRungeKutta(Charge,fMaxStep, vecRKIn, vecRKOut);
+  //  printf(" now at     x=%f  y=%f  z=%f \n", vecRKOut[0],vecRKOut[1],vecRKOut[2]);
+  vecOut[0] = vecRKOut[0];
+  vecOut[1] = vecRKOut[1];
+  vecOut[2] = vecRKOut[2];
+  vecOut[6] = vecRKOut[6];
+  vecOut[3] = vecRKOut[3];
+  vecOut[4] = vecRKOut[4];
+  vecOut[5] = vecRKOut[5];
+
+  return stepLength;
+}
+
+//______________________________________________________________________________
+double FairRKPropagator::OneStepRungeKutta(double charge, double step,
+    double* vect, double* vout)
+{
+
+  // Wrapper to step with method RungeKutta.
+
+  /// ******************************************************************
+  /// *                *
+  /// *  Runge-Kutta method for tracking a particle through a magnetic *
+  /// *  field. Uses Nystroem algorithm (See Handbook Nat. Bur. of   *
+  /// *  Standards, procedure 25.5.20)         *
+  /// *                *
+  /// *  Input parameters            *
+  /// * CHARGE    Particle charge        *
+  /// * STEP    Step size          *
+  /// * VECT    Initial co-ords,direction cosines,momentum   *
+  /// *  Output parameters             *
+  /// * VOUT    Output co-ords,direction cosines,momentum  *
+  /// *  User routine called             *
+  /// * CALL GUFLD(X,F)            *
+  /// *                *
+  /// *    ==>Called by : <USER>, GUSWIM         *
+  /// * Authors    R.Brun, M.Hansroul  *********     *
+  /// *      V.Perevoztchikov (CUT STEP implementation)  *
+  /// *                *
+  /// *                *
+  /// ******************************************************************
+
+  double h2, h4, f[4];
+  double xyzt[3], a=0, b=0, c=0, ph,ph2;
+  double secxs[4],secys[4],seczs[4]; //hxp[3];
+  double /*g1 , g2, g3, g4, g5, g6,*/ ang2, dxt, dyt, dzt;
+  double est, at, bt, ct, cba;
+//  double /*f1, f2, f3, f4, rho, tet, hnorm, hp, rho1, sint, cost*/;
+
+  double x=0;
+  double y=0;
+  double z=0;
+
+  double track_length = 0.;
+
+  double xt;
+  double yt;
+  double zt;
+
+  double maxit = 1992;
+  double maxcut = 11;
+
+  const double hmin   = 1e-4; // !!! MT ADD,  should be member
+  const double kdlt   = 1e-6; // !!! MT CHANGE from 1e-4, should be member
+  const double kdlt32 = kdlt/32.;
+  const double kthird = 1./3.;
+  const double khalf  = 0.5;
+  const double kec    = 2.99792458e-4;
+  const double kpisqua = 9.86960440109;
+  /*  const int kix  = 0;
+    const int kiy  = 1;
+    const int kiz  = 2;
+    const int kipx = 3;
+    const int kipy = 4;
+    const int kipz = 5;
+  */
+  // *.
+  // *.    ------------------------------------------------------------------
+  // *.
+  // *             this constant is for units cm,gev/c and kgauss
+  // *
+  int iter = 0;
+  int ncut = 0;
+  for(int j = 0; j < 7; j++) {
+    vout[j] = vect[j];
+  }
+
+  double pinv   = kec * charge / vect[6];
+  double tl = 0.;
+  double h = step;
+  double rest;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'rest' can be reduced.](https://app.codacy.com/app/FairRootGroup/FairRoot/pullRequest?prid=4139720)